src/ts_catalog/continuous_agg.c

```diff
@@ -1898,3 +1899,38 @@ ts_cagg_permissions_check(Oid cagg_oid, Oid userid)
 
 	return ownerid;
 }
+
+Query *
+ts_continuous_agg_get_query(ContinuousAgg *cagg)
+{
+	Oid cagg_view_oid;
+	Relation cagg_view_rel;
+	RuleLock *cagg_view_rules;
+	RewriteRule *rule;
+	Query *cagg_view_query;
+
+	/*
+	 * Get the direct_view definition for the finalized version because
+	 * the user view doesn't have the "GROUP BY" clause anymore.
+	 */
+	if (ContinuousAggIsFinalized(cagg))
+		cagg_view_oid =
+			get_relname_relid(NameStr(cagg->data.direct_view_name),
+							  get_namespace_oid(NameStr(cagg->data.direct_view_schema), false));
+	else
+		cagg_view_oid =
+			get_relname_relid(NameStr(cagg->data.user_view_name),
+							  get_namespace_oid(NameStr(cagg->data.user_view_schema), false));
+
+	cagg_view_rel = table_open(cagg_view_oid, AccessShareLock);
+	cagg_view_rules = cagg_view_rel->rd_rules;
+	Assert(cagg_view_rules && cagg_view_rules->numLocks == 1);
+
+	rule = cagg_view_rules->rules[0];
+	Ensure(rule->event == CMD_SELECT, "unexpected rule event %d for view", rule->event);
```

Comment:
this should not be ensure but proper error check

Reply:
Ooops... fixed!!!